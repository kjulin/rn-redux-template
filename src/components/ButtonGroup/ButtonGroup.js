import React, {Component, PropTypes} from 'react';
import {StyleSheet, View, Text} from 'react-native';
import Button from '../Button/Button';

export default class ButtonGroup extends Component {
    static propTypes = {
        buttonLabels: PropTypes.arrayOf(PropTypes.string).isRequired,
        onSelect: PropTypes.func.isRequired,
        selected: PropTypes.string
    };

    isActive(value) {
        return this.props.selected === value;
    }

    render() {
        return (
            <View style={this.getStyle('containerStyle')}>
                <Text style={this.getStyle('labelStyle')}>{this.props.label}</Text>
                <View style={styleSheet.buttonContainer}>
                    {this.props.buttonLabels.map((value, index) => {
                        return (
                            <Button
                                key={index}
                                onClick={() => this.props.onSelect(value)}
                                label={value}
                                style={[this.getStyle('buttonStyle').concat(this.isActive(value) ? this.getStyle('activeButtonStyle') : [])]}
                                labelStyle={[this.getStyle('buttonLabelStyle').concat(this.isActive(value) ? this.getStyle('activeButtonLabelStyle') : [])]}/>
                        );
                    })}
                </View>
            </View>
        );
    }

    getStyle(key) {
        let styles = [];
        styles.push(styleSheet[key]);
        if (this.props[key] !== undefined) {
            styles.push(this.props[key]);
        }
        return styles;
    }
}

const styleSheet = StyleSheet.create({
    activeButtonStyle: {
        backgroundColor: '#eeeeff',
        borderWidth: 1,
    },
    activeButtonLabelStyle: {
        color: '#f0f',
    },
    buttonContainer: {
        flexDirection: 'row',
        borderLeftWidth: 1,
        borderColor: '#000',
    },
    buttonStyle: {
        paddingLeft: 12,
        paddingRight: 12,
        paddingTop: 2,
        paddingBottom: 2,
        borderWidth: 1,
        borderRadius: 0,
        borderColor: '#000',
        backgroundColor: '#fff',
        borderLeftWidth: 0,
    },
    buttonLabelStyle: {
        color: '#000'
    },
    labelStyle: {
        color: '#000',
        fontSize: 15,
        marginBottom: 5,
    }
});