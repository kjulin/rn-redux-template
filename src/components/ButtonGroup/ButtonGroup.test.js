import ButtonGroup from './ButtonGroup';
import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('../Button/Button', () => 'Button');

describe('ButtonGroup', () => {
    jest.resetModules();

    const labels = ['one', 'two'];

    it('renders label for group and buttons for each item in buttonLabels array', done => {
        const group = <ButtonGroup label='foo' buttonLabels={labels} onSelect={jest.fn()}/>;
        expect(renderer.create(group)).toMatchSnapshot();
        done();
    });

    // it('highlights button background and label if its value matches group\'s selected prop', done => {
    //     const group = <ButtonGroup buttonLabels={labels} onSelect={jest.fn()} selected={'one'}/>;
    //     expect(renderer.create(group)).toMatchSnapshot();
    //     done();
    // });
});

describe('ButtonGroup DOM', ()=> {
    jest.resetModules();
    const {shallow} = require('enzyme');

    it('triggers onClick callback and passes button\'s value as parameter when pressed', done => {
        const onPressHandler = jest.fn();
        const wrapper = shallow(<ButtonGroup buttonLabels={['one', 'two']} onSelect={onPressHandler}/>);

        wrapper.find('Button[label="one"]').simulate('click');
        wrapper.find('Button[label="two"]').simulate('click');

        expect(onPressHandler).toHaveBeenCalledWith('one');
        expect(onPressHandler).toHaveBeenCalledWith('two');
        expect(onPressHandler).toHaveBeenCalledTimes(2);
        done();
    });
});