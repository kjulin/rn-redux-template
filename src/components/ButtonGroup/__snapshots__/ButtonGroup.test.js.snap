exports[`ButtonGroup renders label for group and buttons for each item in buttonLabels array 1`] = `
<View
  style={
    Array [
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 15,
          "marginBottom": 5,
        },
      ]
    }>
    foo
  </Text>
  <View
    style={
      Object {
        "borderColor": "#000",
        "borderLeftWidth": 1,
        "flexDirection": "row",
      }
    }>
    <Button
      label="one"
      labelStyle={
        Array [
          Array [
            Object {
              "color": "#000",
            },
          ],
        ]
      }
      onClick={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#000",
              "borderLeftWidth": 0,
              "borderRadius": 0,
              "borderWidth": 1,
              "paddingBottom": 2,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 2,
            },
          ],
        ]
      } />
    <Button
      label="two"
      labelStyle={
        Array [
          Array [
            Object {
              "color": "#000",
            },
          ],
        ]
      }
      onClick={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#000",
              "borderLeftWidth": 0,
              "borderRadius": 0,
              "borderWidth": 1,
              "paddingBottom": 2,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 2,
            },
          ],
        ]
      } />
  </View>
</View>
`;
